### PR TITLE
Document new ingest and search pipeline allowlist settings

### PR DESCRIPTION
**Pull request by andrross**
### Description
Document new setting being added in https://github.com/opensearch-project/OpenSearch/issues/14439

### Version
2.16+

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ingest-pipelines/processors/index-processors.md [hunk 1]
```diff
@@ -71,3 +71,7 @@ Processor type | Description
...
+
+## Selectively enabling processors
+
+Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously-enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/index-processors.md:77]:
```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.
```

File: _ingest-pipelines/processors/index-processors.md [hunk 2]
```diff
@@ -71,3 +71,7 @@ Processor type | Description
...
+
+## Selectively enabling processors
+
+Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the static cluster setting `ingest-common.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/index-processors.md:77]:
```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be enabled by providing the ingest-common.processors.allowed cluster setting. If not provided, all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, any pipeline using a disabled processor will fail after node restart when the new setting takes effect.
```

File: _search-plugins/search-pipelines/search-processors.md [hunk 3]
```diff
@@ -121,3 +121,7 @@ The response contains the `search_pipelines` object that lists the available req
 
 In addition to the processors provided by OpenSearch, additional processors may be provided by plugins.
 {: .note}
+
+## Selectively enabling processors
+
+Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) can be selectively enabled by providing the following static cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If not specified, then all processors are enabled by default. If an empty list is specified, then all processors are disabled. If the setting is changed to remove previously-enabled processors, then any pipeline using a disabled processor will fail after the node is restarted and the new setting takes effect.
```
Comment (vagimeli) [inline _search-plugins/search-pipelines/search-processors.md:127]:
```suggestion
Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) are selectively enabled through the following cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If unspecified, all processors are enabled. An empty list disables all. Removing enabled processors causes pipelines using them to fail after node restart.
```

**Review by vagimeli — APPROVED** [review]
Doc review complete. Moving to editorial.

File: _ingest-pipelines/processors/index-processors.md [hunk 4]
```diff
@@ -71,3 +71,7 @@ Processor type | Description
...
+
+## Selectively enabling processors
+
+Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the `ingest-common.processors.allowed` cluster setting. If not provided, all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, any pipeline using a disabled processor will fail after node restart when the new setting takes effect.
```
Comment (natebower) [inline _ingest-pipelines/processors/index-processors.md:77]:
```suggestion
Processors defined by the [ingest-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonPlugin.java) can be selectively enabled by providing the `ingest-common.processors.allowed` cluster setting. If not provided, then all processors are enabled by default. Specifying an empty list disables all processors. If the setting is changed to remove previously enabled processors, then any pipeline using a disabled processor will fail after node restart when the new setting takes effect.
```

File: _search-plugins/search-pipelines/search-processors.md [hunk 5]
```diff
@@ -121,3 +121,7 @@ The response contains the `search_pipelines` object that lists the available req
 
 In addition to the processors provided by OpenSearch, additional processors may be provided by plugins.
 {: .note}
+
+## Selectively enabling processors
+
+Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) are selectively enabled through the following cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If unspecified, all processors are enabled. An empty list disables all. Removing enabled processors causes pipelines using them to fail after node restart.
```
Comment (natebower) [inline _search-plugins/search-pipelines/search-processors.md:127]:
```suggestion
Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) are selectively enabled through the following cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If unspecified, then all processors are enabled. An empty list disables all processors. Removing enabled processors causes pipelines using them to fail after node restart.
```

Comment (natebower) [inline _search-plugins/search-pipelines/search-processors.md:127]: Should "a" precede "node restart"?

**Review by natebower — COMMENTED** [review]
@vagimeli @andrross Just a few changes. Thanks!

File: _ingest-pipelines/processors/index-processors.md (hunk 4 above)
Comment (vagimeli) [reply _ingest-pipelines/processors/index-processors.md:77]: @natebower One day "If, then" will stick for me 💯

File: _search-plugins/search-pipelines/search-processors.md [hunk 6]
```diff
@@ -121,3 +121,7 @@ The response contains the `search_pipelines` object that lists the available req
 
 In addition to the processors provided by OpenSearch, additional processors may be provided by plugins.
 {: .note}
+
+## Selectively enabling processors
+
+Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) are selectively enabled through the following cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If unspecified, then all processors are enabled. An empty list disables all processors. Removing enabled processors causes pipelines using them to fail after node restart.
```
Comment (vagimeli) [inline _search-plugins/search-pipelines/search-processors.md:127]:
```suggestion
Processors defined by the [search-pipeline-common module](https://github.com/opensearch-project/OpenSearch/blob/2.x/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java) are selectively enabled through the following cluster settings: `search.pipeline.common.request.processors.allowed`, `search.pipeline.common.response.processors.allowed`, or `search.pipeline.common.search.phase.results.processors.allowed`. If unspecified, then all processors are enabled. An empty list disables all processors. Removing enabled processors causes pipelines using them to fail after a node restart.
```

**Pull request merged**